### PR TITLE
Fix analysis bugs

### PR DESCRIPTION
**Pull request by MaggieKimani1**
Fixes analysis bugs exposed by Sonarcloud

image

### CODE REVIEW

File: FileService/Services/HttpClientUtility.cs [hunk 1]
```diff
@@ -39,6 +39,11 @@ public async Task<string> ReadFromDocumentAsync(HttpRequestMessage requestMessag
             requestMessage.Method ??= HttpMethod.Get; // default is GET
 
             using var httpResponseMessage = await _httpClient?.SendAsync(requestMessage);
+            if (httpResponseMessage == null)
+            {
+                throw new InvalidOperationException();
```
Comment (baywet) [inline FileService/Services/HttpClientUtility.cs:44]: maybe add a message so we can pinpoint the issue easily?

File: OpenAPIService/OpenApiService.cs [hunk 2]
```diff
@@ -321,7 +321,7 @@ private OpenApiOperation[] GetOpenApiOperations(OpenApiUrlTreeNode rootNode, str
                      * between the relative url segment and the corresponding OpenApiUrlTreeNode segment name
                      * ex.: matching '/users/12345/messages' with '/users/{user-id}/messages'
                      */
-                    tempTargetChild = targetChild.Children
+                    tempTargetChild = targetChild?.Children
```
Comment (baywet) [inline OpenAPIService/OpenApiService.cs:324]: is it possible that children is null when targetChild is not?

File: OpenAPIService/PowershellFormatter.cs [hunk 3]
```diff
@@ -103,20 +103,24 @@ public override void Visit(OpenApiOperation operation)
         /// <param name="schema">The target <see cref="OpenApiSchema"/></param>
         public override void Visit(OpenApiSchema schema)
         {
-            if (_schemaLoop.Contains(schema))
+            if (schema != null)
```
Comment (baywet) [inline OpenAPIService/PowershellFormatter.cs:106]: collapse this if statement with the next one

**Review by baywet — CHANGES_REQUESTED** [review]

File: OpenAPIService/OpenApiService.cs (hunk 2 above)
Comment (irvinesunday) [reply OpenAPIService/OpenApiService.cs:324]: It is possible. We should also add a null-conditional after `Children`

Comment (MaggieKimani1) [reply OpenAPIService/OpenApiService.cs:324]: @irvinesunday

File: OpenAPIService/PowershellFormatter.cs (hunk 3 above)
Comment (MaggieKimani1) [reply OpenAPIService/PowershellFormatter.cs:106]: Is this to help with test coverage?

File: FileService/Services/HttpClientUtility.cs [hunk 4]
```diff
@@ -39,9 +39,14 @@ public async Task<string> ReadFromDocumentAsync(HttpRequestMessage requestMessag
             requestMessage.Method ??= HttpMethod.Get; // default is GET
 
             using var httpResponseMessage = await _httpClient?.SendAsync(requestMessage);
-            var fileContents = await httpResponseMessage?.Content?.ReadAsStringAsync();
+            if (httpResponseMessage == null)
+            {
+                throw new InvalidOperationException("The request message was already sent by the HttpClient instance.");
```
Comment (baywet) [inline FileService/Services/HttpClientUtility.cs:44]: I don't believe this message accurately reflects the situation here.

**Review by baywet — CHANGES_REQUESTED** [review]

File: OpenAPIService/PowershellFormatter.cs (hunk 3 above)
Comment (baywet) [reply OpenAPIService/PowershellFormatter.cs]: no, it just makes less code to maintain, and reading it easier. Thanks for making the change.

**Review by zengin — CHANGES_REQUESTED** [review]

File: FileService/Services/HttpClientUtility.cs [hunk 5]
```diff
@@ -39,9 +39,14 @@ public async Task<string> ReadFromDocumentAsync(HttpRequestMessage requestMessag
...
+            }
+
+            var fileContents = await httpResponseMessage.Content?.ReadAsStringAsync();
 
-            return !httpResponseMessage.IsSuccessStatusCode ? throw new Exception(fileContents) : fileContents;
+            return httpResponseMessage.IsSuccessStatusCode ? throw new Exception(fileContents) : fileContents;
```
[the remark this answers is not in the document]
Comment (MaggieKimani1) [reply FileService/Services/HttpClientUtility.cs:49]: That's correct, thank you for pointing it out.
Resolving...

**Review by baywet — DISMISSED** [review]

File: FileService.Test/HttpClientUtilityShould.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        public HttpClientUtilityShould()
+        {
+            _httpClientMock = new HttpClient(new MockHttpMessageHandler());
+            _httpClientUtility = new HttpClientUtility(_httpClientMock);
+        }
+
+        [Fact]
+        public async Task ReturnContentAsString()
+        {
+            HttpContent httpContent = new StringContent("Content as string");
```
Comment (zengin) [inline FileService.Test/HttpClientUtilityShould.cs:28]: nit: consider defining "Content as string" as constant.

**Review by zengin — DISMISSED** [review]

**Review by baywet — APPROVED** [review]

**Pull request merged**